pkg/pillar/cmd/volumemgr/updatestatus.go

```diff
@@ -592,7 +608,16 @@ func doUpdateVol(ctx *volumemgrContext, status *types.VolumeStatus) (bool, bool)
 			status.State != types.CREATING_VOLUME &&
 			status.SubState == types.VolumeSubStateInitial {
 
-			_, err := ctx.casClient.GetImageHash(ctStatus.ReferenceID())
+			status.IsAppImage = false
+			imgName := ctStatus.ReferenceID()
+			if ctx.hvTypeKube {
+				cfg := lookupContentTreeConfig(ctx, status.ContentID.String())
+				if cfg != nil && cfg.IsAppImage && ctStatus.OciImageName != "" {
+					imgName = ctStatus.OciImageName
+					status.IsAppImage = true
```

Comment:
When you modify status you need to set changed to true so that the function returns true (and the caller hence publishes the status).
The logic here makes it hard to tell whether you modified IsAppImage since you force it to false above.